### DISCUSSION TITLE
Hardware/Software Interface Questions

### DISCUSSION THREAD

**Discussion by flyboyCW12** [iNavFlight/inav#8070] [General]
Hello,

This is a general engineering question aimed at those with much more experience of INAV, Matek FCs and Embedded Systems Development than I currently have.  If I'm in the wrong place to ask for help please let me know with suggestions (hopefully) of where to go next.

I've been using INAV for a couple of year now with a Matek FC (411 Wing) and am planning to add another " ESP 32 - based, master control board" (running Micropython) to an aircraft for various reasons, amongst them the need to transmit a "beacon signal" via a WiFi link to comply with French regulations;  the FC and INAV + the pilot will retain full control of the aircraft flight

At his stage of the development process I'm trying to zero in on the best way to get telemetry data out of the 411!  Options which come straight to mind are I2C and UART based links from the hardware point of view.  Protocols I need to research clearly.

Any initial thoughts/pointers on this would be greatly appreciated. .

Harry

**Comment by stronnag** [comment]
LTM is a simple, lightweight serial protocol which works at low baud rate (suitable for soft serial on a F411).
Well documented in the [wiki](https://github.com/iNavFlight/inav/wiki/Lightweight-Telemetry-(LTM))

**Reply by flyboyCW12** [reply]
Thank you.  I will study this.  Just the sort of pointer I was hoping for.

Harry
